base/compiler/abstractinterpretation.jl

```diff
@@ -203,6 +203,8 @@ function const_prop_profitable(@nospecialize(arg))
             isconstType(b) && return true
             const_prop_profitable(b) && return true
         end
+    elseif isa(arg, PartialOpaque)
```

Comment:
I thought the major benefit of `OpaqueClosure` is that you don't have to aggressively specialize on the value. This seems to be doing precisely the opposite, and aggressively forcing repeated inference on any user of them.